### ISSUE TITLE
Quantization of LeNET model using MNIST dataset breaks during model freeze

### ISSUE THREAD

**Issue by peter197321** [tensorflow/tensorflow#40200]
I'm trying to train Lenet-net using the MNIST dataset from [here](http://yann.lecun.com/exdb/mnist/ ) and to quantize its float model. My steps are the following:

Firstly, I have applied MNIST dataset to train the classifier which works fines. 

```
python3 train_image_classifier.py --dataset_dir=tmp/mnist --dataset_name=mnist --train_dir=train/mnist --model_name=lenet --clone_on_cpu=true --max_number_of_steps=50000 --quantize_delay=40000
```

Secondly, I have exported the trained model to the pb format and it works well.

```
python3 export_inference_graph.py --dataset_dir=tmp/mnist --dataset_name=mnist --train_dir=train/mnist --model_name=lenet --checkpoint=train/mnist/model.ckpt-50000 --quantize --output_file=mnist.pb 
```

Finally, I try to freeze graph using 
```
freeze_graph --input_graph=mnist.pb --input_checkpoint=train/mnist/model.ckpt-50000 --output_graph=mnist_frozen.pb --input_binary=true --output_node_names=Predictions/Reshape_1 
``` 

but the following error occurred

```
InvalidArgumentError: Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

Assign requires shapes of both tensors to match. lhs shape= [5,5,3,32] rhs shape= [5,5,1,32] 
```

Please can you advise how to freeze it correctly?

BTW, I have tried other models like CIFAR10 and all process works fine.

**Comment by amahendrakar** [comment]
@peter197321,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here and also the TensorFlow version you are using. Thanks!

**Comment by peter197321** [comment]
**Tools version:**
Python= 3.7.7, TensorFlow=1.15.3, Numpy=1.16.4, contextlib2=0.6.0, pillow=7.1.2

**Sources:**
**```
https://github.com/tensorflow/models/blob/master/research/slim/train_image_classifier.py
https://github.com/tensorflow/models/blob/master/research/slim/export_inference_graph.py
```**


freeze_graph and tflite_convert tools taken from Anaconda3 environment

**Comment by peter197321** [comment]
I guess I encounter an error in script [export_inference_graph.py](https://github.com/tensorflow/models/blob/57e075203f8fba8d85e6b74f17f63d0a07da233a/research/slim/export_inference_graph.py#L133) where image-grayscale and model are needed so I've changed hard-coded the number of channels to `1 ` for my case.

On another hand, I used to add explicit control switch `--use_grayscale` for a newer version.

Now it seems it's working.
